src/js/Controllers/TTSController.js

```diff
@@ -2,10 +2,80 @@ import RpcFactory from './RpcFactory'
 class TTSController {
     constructor () {
         this.addListener = this.addListener.bind(this)
+        this.audioPlayer = new Audio();
+        this.filePlaylist = [];
     }
     addListener(listener) {
         this.listener = listener
     }
+
+    playAudio() {
+        if(this.filePlaylist.length == 0) {
+            this.audioPlayer.onended = null;
+            if(!this.audioPlayer.paused) {
+                this.audioPlayer.pause();
+                this.audioPlayer.src = "";
+                return;
+            }
+        }
+
+        var path = this.filePlaylist[0].text;
+        this.filePlaylist.shift();
+
+        this.audioPlayer.onended = () => {
+            this.audioPlayer.src ="";
+            if(this.filePlaylist[0]) {
+                if(this.filePlaylist[0].type === "FILE") {
+                    this.playAudio();
+                } else if (this.filePlaylist[0].type === "TEXT"){
+                    this.speak();
+                }
+            }
+        }
```

Comment:
Should have an error function for this as well, sometimes the audio file type is not supported.

Reply:
updated